apiBase/APIRequestUtils.es6.js

```diff
@@ -4,90 +4,54 @@ import ValidationError from './errors/ValidationError';
 import NoModelError from './errors/NoModelError';
 import APIResponse from './APIResponse';
 
-import Events from './Events';
-
-const EventEmitterShim = {
-  emit: () => {},
-  on: () => {},
-  off: () => {},
-};
-
 const DefaultOptions = {
   userAgent: 'snoodev3',
   origin: 'https://www.reddit.com',
   appName: 'node-api-client-v3',
   env: 'develop',
   token: '',
   timeout: 5000,
-  eventEmitter: EventEmitterShim,
-};
-
-export const makeOptions = (overrides={}) => {
-  return {
-    ...DefaultOptions,
-    ...overrides,
-  };
-};
-
-const getEmitter = (apiOptions) => {
-  return (apiOptions.eventEmitter || EventEmitterShim);
-};
-
-const requestAuthHeader = (apiOptions) => {
-  const token = apiOptions.token;
-  if (!token) { return {}; }
-  return { Authorization: `Bearer ${token}` };
-};
-
-const requestHeaders = (apiOptions) => {
-  const authHeaders = requestAuthHeader(apiOptions);
-  return {
-    ...(apiOptions.defaultHeaders || {}),
-    ...authHeaders,
-  };
 };
 
-const requestPath = (apiOptions, path) => {
-  return `${apiOptions.origin}/${path}`;
-};
+export const makeOptions = (overrides={}) => ({ ...DefaultOptions, ...overrides });
 
-const appParameter = (apiOptions) => {
-  return `${apiOptions.appName}-${apiOptions.env}`;
+const requestHeaders = ({ token, defaultHeaders={} }) => {
+  const authHeaders = token ? { Authorization: `Bearer ${token}` } : {};
+  return { ...defaultHeaders, ...authHeaders };
 };
 
-export const rawSend = (apiOptions, method, path, data, kind, cb) => {
-  const origin = apiOptions.origin;
-  const url = requestPath(apiOptions, path);
-
-  const fakeReq = {
-    origin,
-    path,
-    url,
-    method,
-    query: { ...data},
-  };
+export const rawSend = (apiOptions, method, path, data, kind) => {
+  const { origin, appName, env } = apiOptions;
+  const url = `${origin}/${path}`;
 
-  getEmitter(apiOptions).emit(Events.request, fakeReq);
   let s = superagent[method](url);
   s.set(requestHeaders(apiOptions));
 
-  data.app = appParameter(apiOptions);
+  const newData = { ...data, app: `${appName}-${env}` };
 
   if (kind === 'form') {
     s.type('form');
-    s.send(data);
+    s.send(newData);
   } else {
-    s.query(data);
+    s.query(newData);
 
     if (s.redirects) {
       s.redirects(0);
     }
   }
 
-  s.end((err, res) => {
-    // handle super agent inconsistencies
-    const req = res ? res.request : fakeReq;
-    cb(err, res, req);
+  return new Promise((resolve, reject) => {
```

Comment:
yes. We looked into this before. Reading the docs it is a little confusing because it seems to indicate it does, however I found [this issue](https://github.com/visionmedia/superagent/issues/957) which  confirms that they only support promises in 2.0 ( we are on 1.3) 

I believe we found out that one of our other libraries was transforming our code to be compatible with promises ( facebook regenerator lib or something). which seems sketchy. We should update the dep or stick with the how its written now I think.